### ISSUE TITLE
Better handling of warning() during cli_progress_bar

### ISSUE THREAD

**Issue by c-hoffmann** [r-lib/cli#596]
Throwing a warning during the runtime of a cli_progress_bar leaves the warning with leftover traces of the progress bar (such as an old percentage count). Also, after a warning() has been thrown, the cli_progress_bar text is colored in red, while I think it should remain in its original (black) color. If this is implemented this way intentionally, please close my ticket.

![warning_cli](https://user-images.githubusercontent.com/105429299/235360540-bcc69611-7f11-42c9-a05b-2c2a7bfc7414.JPG)

**Comment by gaborcsardi** [comment]
Yes, unfortunately that does not look great, but it is a pretty hard problem to solve, possibly impossible, because cli cannot detect warnings.
